webapi/src/main/scala/org/knora/webapi/responders/v1/OntologyResponderV1.scala

```diff
@@ -412,54 +405,55 @@ final case class OntologyResponderV1Live(
-                                         .map(_.toOntologySchema(InternalSchema))
-                                         .filter { ontologyMetadata =>
-                                           // In V1, the only built-in ontology we show is knora-base.
-                                           val ontologyLabel = ontologyMetadata.ontologyIri.getOntologyName
-                                           ontologyLabel == OntologyConstants.KnoraBase.KnoraBaseOntologyLabel || !OntologyConstants.BuiltInOntologyLabels
-                                             .contains(ontologyLabel)
-                                         }
-                                         .map { ontologyMetadata =>
-                                           val project = projectsMap(ontologyMetadata.projectIri.get.toString)
-
-                                           NamedGraphV1(
-                                             id = ontologyMetadata.ontologyIri.toString,
-                                             shortname = project.shortname,
-                                             longname = project.longname
-                                               .getOrElse(
-                                                 throw InconsistentRepositoryDataException(
-                                                   s"Project ${project.id} has no longname"
-                                                 )
-                                               ),
-                                             description = project.description.headOption
-                                               .getOrElse(
-                                                 throw InconsistentRepositoryDataException(
-                                                   s"Project ${project.id} has no description"
-                                                 )
-                                               )
-                                               .toString,
-                                             project_id = project.id,
-                                             uri = ontologyMetadata.ontologyIri.toString,
-                                             active = project.status
-                                           )
-                                         }
-
-      response = NamedGraphsResponseV1(
-                   vocabularies = namedGraphs
-                 )
+      readOntologyMetadataV2 <-
+        messageRelay
+          .ask[ReadOntologyMetadataV2](
+            OntologyMetadataGetByProjectRequestV2(
+              projectIris = projectIris.map(_.toSmartIri),
+              requestingUser = userProfile
+            )
+          )
+
+      projectsMap =
+        projectsResponse.projects.map { project =>
+          project.id -> project
+        }.toMap
+
+      namedGraphs <-
+        ZIO.foreach {
+          readOntologyMetadataV2.ontologies.toVector
+            .map(_.toOntologySchema(InternalSchema))
```

Comment:
Unfortunately the `SmartIri.toOntologySchema` method is not safe to call, it is throwing different exceptions under multiple circumstances.

I have added `IriConverter.asInternalSmartIri` as a replacement for this particular conversion.

Reply:
This won't work here, but instead for now I wrapped it up with `ZIO.attempt`.